cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectS2ICluster.java

```diff
@@ -97,6 +103,7 @@ public static KafkaConnectS2ICluster fromAssembly(
         KafkaConnectS2ICluster kafkaConnect =  new KafkaConnectS2ICluster(namespace, cluster, Labels.fromResource(dep));
 
         kafkaConnect.setReplicas(dep.getSpec().getReplicas());
+        Container container = dep.getSpec().getTemplate().getSpec().getContainers().get(0);
         kafkaConnect.setHealthCheckInitialDelay(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getReadinessProbe().getInitialDelaySeconds());
         kafkaConnect.setHealthCheckTimeout(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getReadinessProbe().getTimeoutSeconds());
```

Comment:
I was using it for to get the container environment variables: https://github.com/kyguy/strimzi/blob/224cd1101380f6c6f3b6dbf054f2e19315c42645/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectS2ICluster.java#L115 to see if the metrics were enabled. I'll move that line further down so that its more clear

Reply:
You did it good. But what Tom is asking for is to use this also in the other fields ... basically, you should use it also on the other two lines:

```java
Container container = dep.getSpec().getTemplate().getSpec().getContainers().get(0);
kafkaConnect.setHealthCheckInitialDelay(container.getReadinessProbe().getInitialDelaySeconds());
kafkaConnect.setHealthCheckTimeout(container.getReadinessProbe().getTimeoutSeconds());
```

That way we save a lot of getter calls which are done again and again.

Reply:
Ah I understand now, makes sense